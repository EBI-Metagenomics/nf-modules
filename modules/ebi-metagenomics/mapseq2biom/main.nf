
process MAPSEQ2BIOM {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:1.2.11--pyhdfd78af_0' :
        'biocontainers/mgnify-pipelines-toolkit:1.2.11--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(msq)
    tuple path(db_otu), val(db_label)

    output:
    tuple val(meta), path("${meta.id}.txt")         , emit: krona_input
    tuple val(meta), path("${meta.id}.tsv")         , emit: biom_out
    tuple val(meta), path("${meta.id}.notaxid.tsv") , emit: biom_notaxid_out
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    mapseq2biom \
        ${args} \
        --krona ${prefix}.txt \
        --no-tax-id-file ${prefix}.notaxid.tsv \
        --label ${db_label} \
        --query ${msq} \
        --otu-table ${db_otu} \
        --out-file ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}.txt
    touch ${prefix}.notaxid.tsv
    touch ${prefix}.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mapseq2biom: 0.1.1
    END_VERSIONS
    """
}
