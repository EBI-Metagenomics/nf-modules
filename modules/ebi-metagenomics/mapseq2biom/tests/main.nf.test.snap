{
    "test_silva - mseq - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.notaxid.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    "versions.yml:md5,d8270f637a909a58d0070c3abc946c52"
                ],
                "biom_notaxid_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.notaxid.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "biom_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "krona_input": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d8270f637a909a58d0070c3abc946c52"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-11T15:15:47.935539"
    },
    "test_silva - mseq": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,fa3ef8980608011a1ecdd15f93889c0a"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,627ca596232e79fe2777f0e9352df993"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.notaxid.tsv:md5,4b4dc1c98f5da99339fa477dd47f371d"
                    ]
                ],
                "3": [
                    "versions.yml:md5,762178a94f67dbd9ae40cbc43b14ba62"
                ],
                "biom_notaxid_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.notaxid.tsv:md5,4b4dc1c98f5da99339fa477dd47f371d"
                    ]
                ],
                "biom_out": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.tsv:md5,627ca596232e79fe2777f0e9352df993"
                    ]
                ],
                "krona_input": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.txt:md5,fa3ef8980608011a1ecdd15f93889c0a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,762178a94f67dbd9ae40cbc43b14ba62"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T13:41:03.879197"
    }
}